Backend: Integrate the formatted email structure with the sorter.

Luna-Jia opened the pull request:
- Develop a mechanism to batch and feed emails to the sorter.
-  Develop integration tests to verify the ETL process from email collection to sorting.

issue #30

Luna-Jia committed:
Backend: Integrate the formatted email structure with the sorter.

- Develop a mechanism to batch and feed emails to the sorter.
-  Develop integration tests to verify the ETL process from email collection to sorting.

issue #30

ghost commented:
Okay, I wrote this in discord but am adding a copy here:

Integration:

- your integration test of email_fetch_real_service works well.
- There is no integration test for email_fetch -> get header/body -> send to sorter -> get real response, at least that I can see. You could call it email_real_sort_test.dart.

MOCK:

- your test email_sorter_running_test works well, however it is a MOCK test and should go under that folder.


so those are the three modifications I need to see.
the email_real_sort_test.dart is the deliverable for dr. jiang.

Luna-Jia committed:
Test: creat a full integration test

- Creat a full integration test "email_real_sort_test.dart" for email_fetch -> get subject/body -> send to sorter -> get real response
- Move mock tests to MOCK folder

issue #30

ghost commented:
Looks fine, if you want to impress dr. jiang I recommend sending some realistic emails from your email address to the jarvis test email account, I'd recommend some of the enron emails, or just copies of less important emails you have.
